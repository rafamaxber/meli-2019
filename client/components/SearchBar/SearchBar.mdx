---
name: SearchBar
menu: Components
---

import { Playground, Props } from 'docz'
import { MainThemeProvider } from '../Layouts/ThemeProvider.tsx'

import SearchBar from './SearchBar.tsx'

# SearchBar

## Properties

<Props of={SearchBar} />

## Basic usage

<Playground>
  <MainThemeProvider>
    <SearchBar />
  </MainThemeProvider>
</Playground>

