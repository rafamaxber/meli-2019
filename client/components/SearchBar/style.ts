import styled from 'styled-components';

export const SearchBar = styled.div`
  display: inline-block;
  max-width: 900px;
  vertical-align: middle;
  left: 26px;
  width: 90%;
  position: relative;
`;

export const Form = styled.form``;

export const Input = styled.input`
  height: 35px;
  padding: 10px;
  width: 95.5%;
  border: none;
  border-radius: ${({ theme }) => `${theme.borderRadius} 0 0 ${theme.borderRadius}`};

  ::placeholder {
    font-size: 18px;
    color: ${({ theme }) => theme.lv3Color};
    padding-top: 2px;
    position: relative;
  }

  :focus {
    outline: none;
    box-shadow: -2px 1px 10px -5px ${({ theme }) => theme.lv1Color};
  }

  @media (max-width: 939px) {
    width: 85%;
  }
`;

export const Button = styled.button`
  border: none;
  cursor: pointer;
  background-color: ${({ theme }) => theme.bgColor};
  border-radius: ${({ theme }) => `0 ${theme.borderRadius} ${theme.borderRadius} 0`};
  padding: 10px;
  display: inline-block;
  vertical-align: top;
  font-size: 0;
  height: 35px;
  position: relative;
`;

export const Typography = styled.span``;
