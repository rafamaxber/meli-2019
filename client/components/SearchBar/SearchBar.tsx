import * as Styled from './style'

const SearchIcon = () => (
  <img src="/static/img/ic_Search.png" />
)

const SearchBar = () => (
  <Styled.SearchBar>
    <Styled.Form action="/items" method="GET" role="search">
      <Styled.Input
        data-testid="search"
        maxLength="120"
        name="search"
        aria-required="true"
        type="search"
        placeholder="Buscar Producto"
        autoCapitalize="off"
        autoCorrect="off"
        aria-autocomplete="list"
        required
        autoFocus
      />
      <Styled.Button type="submit" aria-label="Buscar">
        <SearchIcon />
        <Styled.Typography hidden>Buscar</Styled.Typography>
      </Styled.Button>
    </Styled.Form>
  </Styled.SearchBar>
)

export default SearchBar
