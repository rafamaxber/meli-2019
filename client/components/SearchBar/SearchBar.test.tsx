import React from 'react';
import { render } from '@testing-library/react';

import SearchBar from './SearchBar';

describe('SearchBar', () => {
  describe('smoke tests', () => {
    test('should render component with SearchBar', () => {
      const component = render(<SearchBar/>);

      expect(component).toMatchSnapshot();
    })
  });
})


