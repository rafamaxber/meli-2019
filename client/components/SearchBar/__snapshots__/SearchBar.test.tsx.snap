// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar smoke tests should render component with SearchBar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: inline-block;
  max-width: 900px;
  vertical-align: middle;
  left: 26px;
  width: 90%;
  position: relative;
}

.c1 {
  height: 35px;
  padding: 10px;
  width: 95.5%;
  border: none;
  border-radius: undefined 0 0 undefined;
}

.c1::-webkit-input-placeholder {
  font-size: 18px;
  padding-top: 2px;
  position: relative;
}

.c1::-moz-placeholder {
  font-size: 18px;
  padding-top: 2px;
  position: relative;
}

.c1:-ms-input-placeholder {
  font-size: 18px;
  padding-top: 2px;
  position: relative;
}

.c1::placeholder {
  font-size: 18px;
  padding-top: 2px;
  position: relative;
}

.c1:focus {
  outline: none;
  box-shadow: -2px 1px 10px -5px;
}

.c2 {
  border: none;
  cursor: pointer;
  border-radius: 0 undefined undefined 0;
  padding: 10px;
  display: inline-block;
  vertical-align: top;
  font-size: 0;
  height: 35px;
  position: relative;
}

@media (max-width:939px) {
  .c1 {
    width: 85%;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <form
          action="/items"
          class=""
          method="GET"
          role="search"
        >
          <input
            aria-autocomplete="list"
            aria-required="true"
            autocapitalize="off"
            autocorrect="off"
            class="c1"
            data-testid="search"
            maxlength="120"
            name="search"
            placeholder="Buscar Producto"
            required=""
            type="search"
          />
          <button
            aria-label="Buscar"
            class="c2"
            type="submit"
          >
            <img
              src="/static/img/ic_Search.png"
            />
            <span
              class=""
              hidden=""
            >
              Buscar
            </span>
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="style__SearchBar-sc-286i50-0 dVhlOI"
    >
      <form
        action="/items"
        class="style__Form-sc-286i50-1 bxazgb"
        method="GET"
        role="search"
      >
        <input
          aria-autocomplete="list"
          aria-required="true"
          autocapitalize="off"
          autocorrect="off"
          class="style__Input-sc-286i50-2 fogGJj"
          data-testid="search"
          maxlength="120"
          name="search"
          placeholder="Buscar Producto"
          required=""
          type="search"
        />
        <button
          aria-label="Buscar"
          class="style__Button-sc-286i50-3 kLGETb"
          type="submit"
        >
          <img
            src="/static/img/ic_Search.png"
          />
          <span
            class="style__Typography-sc-286i50-4 daqenF"
            hidden=""
          >
            Buscar
          </span>
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
